pkg/operator/ceph/controller/spec_test.go

```diff
@@ -373,3 +373,17 @@ func TestConfigureExternalMetricsEndpoint(t *testing.T) {
 		assert.Equal(t, "192.168.0.1", currentEndpoints.Subsets[0].Addresses[0].IP, currentEndpoints)
 	})
 }
+
+func TestLogCollectorContainer(t *testing.T) {
+	maxsize, _ := resource.ParseQuantity("1Mi")
+	c := cephv1.ClusterSpec{
+		LogCollector: cephv1.LogCollectorSpec{
+			Enabled:     true,
+			Periodicity: "1d",
+			MaxLogSize:  &maxsize,
+		},
+	}
+	script := fmt.Sprintf(cronLogRotate, "ceph-mon-a", "daily", "1M")
+	r := LogCollectorContainer("ceph-mon-a", "rook-ceph", c)
+	assert.Equal(t, script, r.Command[5])
+}
```

Comment:
```suggestion
func TestLogCollectorContainer(t *testing.T) {
	maxsize, _ := resource.ParseQuantity("1Mi")
	c := cephv1.ClusterSpec{
		LogCollector: cephv1.LogCollectorSpec{
			Enabled:     true,
			Periodicity: "1d",
			MaxLogSize:  &maxsize,
		},
	}
	script := fmt.Sprintf(cronLogRotate, "ceph-mon-a", "daily", "1M")
	r := LogCollectorContainer("ceph-mon-a", "rook-ceph", c)
	assert.Equal(t, script, r.Command[5])
}
```

@travisn also added this test to confirm the translation form `resource.Quantity` to bytes

Reply:
How about splitting out the max log size and periodicity conversions to small helper methods, then you can more easily unit test the results and test all the possible combinations. This test is good, but it only tests a single combination.

Reply:
done